packages/api-gateway/src/services/tenantService.ts

```diff
@@ -36,3 +41,119 @@ export async function getOrganization(
 }
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export function tenantServiceBuilder(
+  tenantProcessClient: TenantProcessClient,
+  attributeProcessClient: AttributeProcessClient,
+  catalogProcessClient: CatalogProcessClient
+) {
+  return {
+    getOrganization: async (
+      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      tenantId: tenantApi.Tenant["id"]
+    ): Promise<apiGatewayApi.Organization> => {
+      logger.info(`Retrieving tenant ${tenantId}`);
+
+      return getOrganization(
+        tenantProcessClient,
+        attributeProcessClient,
+        headers,
+        tenantId
+      );
+    },
+    getOrganizationEservices: async (
+      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      params: {
+        origin: tenantApi.ExternalId["origin"];
+        externalId: tenantApi.ExternalId["value"];
+        attributeOrigin: string;
+        attributeCode: string;
+      }
+    ): Promise<apiGatewayApi.EServices> => {
+      const { origin, externalId, attributeOrigin, attributeCode } = params;
+      logger.info(
+        `Retrieving Organization EServices for origin ${origin} externalId ${externalId} attributeOrigin ${attributeOrigin} attributeCode ${attributeCode}`
+      );
+
+      const tenant = await tenantProcessClient.tenant.getTenantByExternalId({
+        headers,
+        params: {
+          origin,
+          code: externalId,
+        },
+      });
+
+      const attribute =
+        await attributeProcessClient.getAttributeByOriginAndCode({
+          headers,
+          params: {
+            origin: attributeOrigin,
+            code: attributeCode,
+          },
+        });
+
+      const allEservices = await getAllEservices(
+        catalogProcessClient,
+        headers,
+        tenant.id,
+        attribute.id
+      );
+      const allowedEservices = allEservices.filter(
+        (eservice) => eservice.descriptors.length > 0
+      );
+
+      const eservices = await Promise.all(
+        allowedEservices.map((eservice) =>
+          enhanceEservice(
+            tenantProcessClient,
+            attributeProcessClient,
+            headers,
+            eservice
+          )
+        )
+      );
+      return { eservices };
+    },
+    revokeTenantAttribute: async (
+      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      params: {
+        origin: tenantApi.ExternalId["origin"];
+        externalId: tenantApi.ExternalId["value"];
+        attributeCode: tenantApi.M2MAttributeSeed["code"];
+      }
+    ): Promise<void> => {
+      const { origin, externalId, attributeCode } = params;
+      logger.info(
+        `Revoking attribute ${attributeCode} of tenant (${origin},${externalId})`
+      );
+
+      await tenantProcessClient.m2m.m2mRevokeAttribute(undefined, {
+        headers,
+        params: {
+          origin,
+          externalId,
+          code: attributeCode,
+        },
+      });
+    },
+    upsertTenant: async (
```

Comment:
This is implemented already including the modifications requested in https://pagopa.atlassian.net/browse/IMN-815

See discussion in the Jira task: all the checks are performed by the tenant process, here in the BFF it is sufficient to call the m2mUpsertTenant as it was done in Scala, skipping the call to the party registry proxy service which is not needed anymore